lib/bridge/IrcHandler.js

```diff
@@ -538,15 +538,82 @@ IrcHandler.prototype.onMode = Promise.coroutine(function*(req, server, channel,
                                                 mode, enabled, arg) {
     let privateModes = ["k", "i", "s"];
 
-    if (privateModes.indexOf(mode) === -1) {
-        return; // ignore everything but privateModes
-    }
     req.log.info(
         "onMode(%s) in %s by %s (arg=%s)",
         (enabled ? ("+" + mode) : ("-" + mode)),
         channel, by, arg
     );
 
+    if (privateModes.indexOf(mode) !== -1) {
+        req.log.info("Private mode");
```

Comment:
This isn't helpful. We can already infer that we will branch here because we log the `mode`  just above on `:541`.